apps/api/src/index.ts

```diff
@@ -88,12 +88,24 @@ app.set("trust proxy", true);
 app.use(
   cors({
     origin: function (origin, fn) {
-      if (
-        !origin ||
-        config.app.url === origin ||
-        // Note: the frontend app URL above should already be included in the `corsOrigins` config
-        config.app.corsOrigins.includes(origin)
-      ) {
+      // Allow any localhost port in development mode
+      if (config.server.nodeEnv === "development") {
```

Comment:
1. we check for `NODE_ENV=development` so that `NODE_ENV=test` can use the `DOMAIN` variable in this CORS logic. (note: `DOMAIN=.example.com`)
2. if dev mode, allow any localhost origin
3. if not dev mode, enforce the origin is the primary or subdomain

the docs link below outlines the regex pattern:
> the pattern `/example\.com$/` will reflect any request that is coming from an origin ending with “example.com”.